HW_1/myproject/myapp/views.py

```diff
@@ -24,23 +34,73 @@ def wrapper(*args, **kwargs):
         return result
     return wrapper
 
-
-def home(request):
-    logger.info('Index page accessed')
-    return HttpResponse("<h1>Страница приложения пиццерии Modernissimo<h1>")
-
-
-
-@log_user_visiting
-def about(request):
-    return HttpResponse(f"<h1>Наше заведение работает с 1999 года</h1>")
+    
+def view_for(request):
+    my_list = ['apple', 'banana', 'orange']   
+    my_dict = {
+        'каждый': 'красный',
+        'охотник': 'оранжевый',
+        'желает': 'желтый',
+        'знать': 'зеленый',
+        'где': 'голубой',
+        'сидит': 'синий',
+        'фазан': 'фиолетовый',
+    } 
+    context = {'my_list': my_list, 'my_dict': my_dict}
+    return render(request, 'myapp/template_for.html', context)
+
+
+def user_orders(request, user_id):
+    user = get_object_or_404(User, pk=user_id)
+    orders = Order.objects.filter(customer=user).order_by('-date_ordered')
+    dict_products_in_order = dict()
+
+    for order in orders:
+        lst_products = []
+        for product in order.products.filter(order=order):
+           lst_products.append(product)
+        dict_products_in_order[order.date_ordered] = lst_products
+        
+    context = {'user': user, 'orders': orders, 'dict_products_in_order': dict_products_in_order}
+    return render(request, 'myapp/user_orders.html', context)
+    
+
+
+def user_orders_sort_time(request, user_id):
+
+    from datetime import datetime
+
+    user = get_object_or_404(User, pk=user_id)
+    orders = Order.objects.filter(customer=user)
+    set_week, set_month, set_year = set(), set(), set()
+    today = datetime.now()
+
+    for order in orders:
+        delta = (today - datetime.fromisoformat(str(order.date_ordered)[:10])).days
+        if delta <= 7:
+            for product in order.products.all():
+                set_week.add(product.name)
+                set_month.add(product.name)
+                set_year.add(product.name)
+        elif delta <= 30:
+            for product in order.products.all():
+                set_month.add(product.name)
+                set_year.add(product.name)
+        elif delta <= 365:
+            for product in order.products.all():
+                set_year.add(product.name)
```

Comment:
Можно сделать значительно проще
```suggestion
    last_week = date.today() - timedelta(days=7)
    last_month = date.today() - timedelta(days=30)
    last_year = date.today() - timedelta(days=365)
    
    set_week = Product.objects.filter(order__customer=user, order__date_ordered__gte=last_week).distinct()
    set_month = Product.objects.filter(order__customer=user, order__date_ordered__gte=last_month).distinct()
    set_year = Product.objects.filter(order__customer=user, order__date_ordered__gte=last_year).distinct()
```

Reply:
Спасибо за обратную связь!